### PR TITLE
8264125: Specification of Taskbar::getIconImage doesn't mention that the returned image might not be equal to the Taskbar::setIconImage one. (eg on Mac OS)

### PR DESCRIPTION
**Pull request by azvegint**
This fix is to explicitly specify that `Taskbar::getIconImage` may return an object different from passed to `Taskbar::setIconImage`.

Actually it is always returns a different object on macOS(the only OS which supports this feature), but I want to save some options if we decide to rework this.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264125](https://bugs.openjdk.java.net/browse/JDK-8264125): Specification of Taskbar::getIconImage doesn't mention that the returned image might not be equal to the Taskbar::setIconImage one. (eg on Mac OS)

### Reviewers
 * [Alexander Zuev](https://openjdk.java.net/census#kizune) (@azuev-java - **Reviewer**) ⚠️ Review applies to 74bccd438409601a954c614d3bb02079abb3312f
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3250/head:pull/3250` \
`$ git checkout pull/3250`

Update a local copy of the PR: \
`$ git checkout pull/3250` \
`$ git pull https://git.openjdk.java.net/jdk pull/3250/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3250`

View PR using the GUI difftool: \
`$ git pr show -t 3250`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3250.diff">https://git.openjdk.java.net/jdk/pull/3250.diff</a>

</details>

### CODE REVIEW

**Review by azuev-java — APPROVED** [review]

**Comment by azvegint** [comment]
CSR added.

**Comment by azvegint** [comment]
> Leaving aside specification. what does an app really want this to return ?
> Should we start from there ?

We are returning current dock icon image from system.

> Do we definitely not want to just hang on to whatever an application set so we can return it without modification ?
> What are the problems with that ?

It might be a good solution, but I see some difficulties here.
If we just save passed image instance, its content might be modified by user between `setIconImage()` and `getIconImage()` calls.

`setIconImage()` internally converts passed image, so it does not follow its changes.
Thus we will get a discrepancy between result from `getIconImage()` and actual dock icon image on macOS side.
(Probably we can mitigate this by modifying the javadoc with some vague wording though.)

Ideally we should make a copy of an icon passed to `setIconImage()` method to prevent modification of its content.
`setIconImage()`/`getIconImage()` works with too wide type `java.awt.Image`, and I am not aware of any reliable way to clone every possible subtype of it.

**Comment by azvegint** [comment]
@mrserb @dbessono please review

File: src/java.desktop/share/classes/java/awt/Taskbar.java [hunk 1]
```diff
@@ -322,7 +322,7 @@ public PopupMenu getMenu() {
     }
 
     /**
-     * Changes this application's icon to the provided image.
+     * Suggests the system to change this application's icon to the provided {@code image}.
```
Comment (prrace) [inline src/java.desktop/share/classes/java/awt/Taskbar.java:325]: Suggests -> Requests

File: src/java.desktop/share/classes/java/awt/Taskbar.java [hunk 2]
```diff
@@ -338,6 +338,10 @@ public void setIconImage(final Image image) {
 
     /**
      * Obtains an image of this application's icon.
+     * <p>
```
Comment (prrace) [inline src/java.desktop/share/classes/java/awt/Taskbar.java:341]: As discussed off-line, preface this with "@apiNote"

Comment (azvegint) [reply src/java.desktop/share/classes/java/awt/Taskbar.java]: updated

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**